addons/web/static/src/views/fields/translation_dialog.js

```diff
@@ -93,11 +104,65 @@ export class TranslationDialog extends Component {
             [this.props.resId],
             this.props.fieldName,
             translations,
+            resetLangs,
         ]);
 
         await this.props.onSave();
         this.props.close();
     }
+
+    onUpdate(term, ev) {
+        const newValue = ev.target.value;
+        if (!this.checkValue(term, newValue)) {
+            ev.target.value = term.value;
+            return;
+        }
+        if (newValue) {
+            term.isModified = true;
+            term.isTranslated = true;
+            term.value = newValue;
+            if (term.lang === "en_US") {
+                // update source and other fallback value
+                this.state.terms.forEach((t) => {
+                    if (t.source === term.source) {
+                        t.source = newValue;
+                        if (!t.isTranslated) {
+                            t.value = newValue;
+                        }
+                    }
+                });
+            }
+        } else {
+            term.isModified = term.isModified || term.isTranslated;
+            term.isTranslated = false;
+            ev.target.value = term.value = term.source;
+        }
+    }
+
+    checkValue(term, value) {
+        if (
+            value &&
+            term.lang === "en_US" &&
+            this.state.terms.some(
+                (t) => t.value === value && t.id != term.id && t.lang === term.lang
+            )
+        ) {
+            this.props.addDialog(AlertDialog, {
+                body: _t("Two sources cannot be the same"),
+            });
+            return false;
+        }
```

Comment:
I don't see a way to reproduce this. Can you help me reproduce? If it can be reproduced, aren't we too restrictive?

Reply:
Reproduce:
Go settings -> Technical -> Views
open one of them (the form view of `ir.ui.view`)
open the translation dialog, usually there should be more than 1 term for each language, and you can make 2 en_US translation value become the same

1. it is a translation dialog, making 2 sources become the same actually changes the content of one of them, which is not in the scope of `translating`
2. more corner cases, one source term has 2 rows, and 2 different translation values

Strictly speaking for the backend, we expect value_en, value_lang has a one to one mapping. I even want to add this check for all languages

Reply:
> making 2 sources become the same actually changes the content of one of them, which is not in the scope of translating

If field value modification is not the scope, why don't we make it readonly? Because at this point, if I changed one of the english terms, I noticed that this change will be reflected in the form. So practically, the translation dialog acts as a wizard to change the value of the field.

Reply:
Because en_US may not be the main language of the company
For example, a French employee write thes value in French `<div>Pomme</div>`
the translation dialog will have terms

1. lang: French, source: `Pomme`, value: `Pomme`
2. lang: English, source: `Pomme`, value: `Pomme`

At this time, the user want to translate `Pomme` to `Apple` for English
And he should expect 

1. lang: French, source: `Apple`, value: `Pomme`
2. lang: English, source: `Apple`, value: `Apple`